Essentials/src/com/earth2me/essentials/RandomTeleport.java

```diff
@@ -175,6 +175,36 @@ public void cacheRandomLocations(Location center, double minRange, double maxRan
         return future;
     }
 
+    /**
+     * Returns an int array of length 2, representing a random coordinate pair that is both inside the square
+     * <pre>{@code x = maxRange * 2 and y = maxRange * 2 centered on (0,0)}</pre>
+     * and also NOT inside the square
+     * <pre>{@code x = minRange * 2 and y = minRange * 2 centered on (0,0)}</pre>
+     * The points generated by this method are evenly distributed throughout the area.
+     *
+     * @param minRange The minimum that {@code |x|} or {@code |y|} will be.
+     * @param maxRange The maximum that {@code |x|} or {@code |y|} will be.
+     * @return A pair of numbers, each randomly generated, that satisfy the above conditions.
+     */
+    private double[] getRandXySquare(double minRange, double maxRange) {
+        double[] randPairAtSize = new double[]{
+                RANDOM.nextDouble() * (maxRange - minRange) + minRange,
+                RANDOM.nextDouble() * (maxRange + minRange) - minRange};
+        switch ((int) (RANDOM.nextDouble() * 4)) {
+            case 0:
+                return randPairAtSize;
+            case 1:
+                return new double[]{-randPairAtSize[1], randPairAtSize[0]};
+            case 2:
+                return new double[]{-randPairAtSize[0], -randPairAtSize[1]};
+            case 3:
+                return new double[]{randPairAtSize[1], -randPairAtSize[0]};
+        }
+        //This line will never be reached as the random number in the switch will ALWAYS be in the range 0-3 inclusive.
+        return new double[]{0, 0};
```

Comment:
Why not just case 1,2,3 and default? Default can catch 0 and any impossible case.

Reply:
Oh true, I like that